File: Pset1/mario.c
#include <cs50.h>
#include <stdio.h>

//print n hashes
void hash(int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("#");
    }
}
//print n spaces
void space(int n)
{
    for (int i = 0; i < n; i++)
    {
        printf(" ");
    }
}

//main code
int main(void)
{
    int h = 0;
    //get height
    while (h > 8 || h < 1)
    {
        h = get_int("Height: ");
    }
    //print out pyramid
    for (int i = 1; i < h + 1; i++)
    {
        space(h - i);
        hash(i);
        space(2);
        hash(i);
        printf("\n");
        //next row
    }
}

File: pset6/mario/more/mario.py
from cs50 import get_int


def main():
    while True:
        height = get_int("Height: ")
        if height > 0 and height <= 8:
            break
        
    for i in range(1, height + 1):
        spaces(height - i)
        hashes(i)
        spaces(2)
        hashes(i)
        print()


# print n number of hashes
def hashes(n):
    print("#" * n, end='')

# print n number of spaces


def spaces(n):
    print(" " * n, end='')


main()
